tests/lcb/stdlib/foreign.lcb

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright (C) 2015 Runtime Revolution Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+module com.livecode.foreign.tests
+
+use com.livecode.foreign
+use com.livecode.__INTERNAL._testlib
+
+handler TestZStringNative_Null()
+	variable t as ZStringNative
+	put "\u{0}" into t
+end handler
+public handler TestZStringNative()
+	-- bug 14826
+	-- Check that strings containing embedded nuls can't be
+	-- bridged.
+	MCUnitTestHandlerThrowsBroken(TestZStringNative_Null, "ZStringNative (nulls", "bug 14829")
+end handler
+
+handler TestZStringUTF16_Null()
+	variable t as ZStringUTF16
+	put "\u{0}" into t
+end handler
+public handler TestZStringUTF16()
+	variable tUTF16 as ZStringUTF16
+	variable tInternal as String
+
+	-- Test round-trip bridging
+	put "�" into tUTF16
```

Comment:
This comes up as '�' on github... What character is it?

I wonder if we should restrict ourselves to ASCII for source-files in general...